### ISSUE TITLE
Feature Request: REST API integration

### ISSUE THREAD

**Issue by papaiatis** [jenkinsci/violations-plugin#42]
Hi,
I've been using this plugin for a while now and our team is thinking about developing a remote monitor service with a dashboard. This dashboard would gather build and code analysis information from multiple sources including Jenkins with Violation plugin. By default, Jenkins support a REST API in different response formats (XML, JSON). However this API is not available under the Violations plugin, for example: http://myjenkins/job/FxCop/4/violations/api/xml.
It would be such a great help to make this URL available.
Thank you.

**Comment by sh1mmer** [comment]
:+1:

**Comment by LaurentTardif** [comment]
:+1:

**Comment by arnlaugsson** [comment]
:+1:

**Comment by GBeauny** [comment]
:+1:

**Comment by GBeauny** [comment]
Something has been done about that issue?
Thank you guys.

**Comment by papaiatis** [comment]
I don't think so... Does not seem to be under maintenance for years.

**Comment by GBeauny** [comment]
Nobody knows an another way to get clover php or cobertura result through Jenkins APIs? 
I'm trying to find a way about it. 

Thanks,

**Comment by tomasbjerre** [comment]
@GBeauny Perhaps you can just do a simple GET request to the Cobertura report file in the workspace "/jenkins/job/jobName/ws/"? But that would only get you the file for the latest build.

**Comment by GBeauny** [comment]
Yes, this a last of last solution, but it's not really nice. But thanks, i may update directly the plugin.

**Comment by ericdriggs** [comment]
I agree that an api supporting json / xml would be helpful. If you're just trying to get cobertura data, you can go directly to cobertura for that info. 

For cobertura, you can get the report as such:
JSON:  /jenkins/job/jobName/buildNumber/cobertura/api/json?pretty=true&depth=3
XML:   /jenkins/job/jobName/buildNumber/cobertura/api/xml?pretty=true&depth=3

Example:
https ://fakeserver.com/jenkins/job/foo/123/cobertura/api/json?pretty=true&depth=3
